src/ddscxx/include/org/eclipse/cyclonedds/domain/qos/DomainParticipantQosDelegate.hpp

```diff
@@ -20,6 +20,8 @@
 
 #include <dds/core/policy/CorePolicy.hpp>
 
+#include <dds/ddsi/ddsi_xqos.h>
```

Comment:
I suspect this header file gets included by the application code as well, and if so, then including this header file pollutes the name space rather badly. It isn't really part of the public API, but language bindings are a bit special, and this does save quite a few lines of boilerplate compared to calling `dds_qget_` functions all the time and checking the return value.

The only thing you actually use from it (unless I overlooked something) are the `QP_` flags, so a the name space pollution could be addressed by reorganising the C header files a bit, but I think I'd be happier if the C++ binding used its own set of (namespaced) constants, with a single `.cpp` file including the C header file using static assertions to verify that the bits are actually the same.

Best would of course be if there's a way of decoupling it completely without adding a lot of boilerplate.